Removed unused code

friscoMad opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
After finding an unused function in code, I did a full static analisys using Vulture: https://pypi.python.org/pypi/vulture

Reviewed line by line unused code and just deleted real dead code and just left as it was everything else.
Deleted:

- Unused exception class
- Visible_forts, get_quartile and clock_extend functions and statics in model.py
- Unused attributes and dead code in schedulers.
- Unused class and function in utils (maybe someone was using this for debugging)

This will provide really minor cpu and memory usage improvements but is mostly to keep the codebase clean.

## Motivation and Context
Just improve code quality

## How Has This Been Tested?
Tested in a small instance, and running on my main instance.

## Screenshots (if appropriate):

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

FrostTheFox commented:
Please fix Travis errors.

Arengo commented:
Travis errors by itself, it seems.

Alderon86 commented:
`ImportError: No module named Cython.Build`
Same error for me, i dont think thats related to the code

sebastienvercammen commented:
I took a look at why Travis is failing, and you can ignore this for now. It's unrelated to your code; I've contacted the author of the dependency that's causing it, it'll be fixed soon.

sebastienvercammen commented on pogom/models.py:
```diff
@@ -28,9 +28,8 @@
-    get_move_name, get_move_damage, get_move_energy, get_move_type, \
-    clear_dict_response
+    get_args, cellid, in_radius, date_secs, clock_between, get_move_name, \
+    get_move_damage, get_move_energy, get_move_type, clear_dict_response
```
For consistency, this should be updated to the multi-line format:
```
from .utils import (x, y,
    z)
```
instead of using the escape character `\`.

jagauthier commented:
I went through this and confirmed that code was not used.  I assume you probably used a python analysis tool that determined these things were not used after running for awhile.  But, i wanted to be thorough, so I also went through and verified.
 
Currently, this PR is in conflict with develop. I am able to run it as it is by changing
`db_schema_version = 16`
to
`db_schema_version = 18`
on models.py

It seems to be fine, but I don't feel comfortable approving until I can test this merged with the current develop.  Will you update it, so I can give it a thorough run through?

friscoMad commented:
Yup I used vulture, a static analysis tool.
I will update this pr later this week, I am on holidays until tomorrow
night.

El lun., 1 de may. de 2017 10:54 PM, jagauthier <notifications@github.com>
escribió:

> I went through this and confirmed that code was not used. I assume you
> probably used a python analysis tool that determined these things were not
> used after running for awhile. But, i wanted to be thorough, so I also went
> through and verified.
>
> Currently, this PR is in conflict with develop. I am able to run it as it
> is by changing
> db_schema_version = 16
> to
> db_schema_version = 18
> on models.py
>
> It seems to be fine, but I don't feel comfortable approving until I can
> test this merged with the current develop. Will you update it, so I can
> give it a thorough run through?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RocketMap/RocketMap/pull/1974#issuecomment-298429818>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAduugxX7FTKB2Ty96_2dcohNSct12sJks5r1kZsgaJpZM4M-dt0>
> .
>

friscoMad commented:
Rebased and removed some more unused code.

@sebastienvercammen Fixed imports to follow coding style

jagauthier commented:
Thanks for rebasing.  I cleaned my tree, pulled this one down and started it up. I let it run for a few hours.  I did not notice anything out of the ordinary.  Of course, this was expected due to the fact that this code was not being used.

Obihoernchen commented on pogom/schedulers.py:
```diff
@@ -996,7 +981,6 @@ def next_item(self, status):
                 log.debug('Enumerating queue found best location: %s.',
                           repr(best))
 
-            prefix = 'Calc %.2f for %d scans:' % (time.time() - now_time, n)
```
Deleting this line will make `n` unused as well, so you can the following two lines as well:
```diff
diff --git a/pogom/schedulers.py b/pogom/schedulers.py
index f5c4440..5297521 100644
--- a/pogom/schedulers.py
+++ b/pogom/schedulers.py
@@ -846,7 +846,6 @@ class SpeedScan(HexSearch):
                 time.sleep(1)
 
             now_date = datetime.utcnow()
-            n = 0  # count valid scans reviewed
             q = self.queues[0]
             ms = ((now_date - self.refresh_date).total_seconds() +
                   self.refresh_ms)
@@ -946,8 +945,6 @@ class SpeedScan(HexSearch):
                     count_late += 1
                     continue
 
-                n += 1
-
                 # Bands are top priority to find new spawns first
                 score = 1e12 if item['kind'] == 'band' else (
                     1e6 if item['kind'] == 'TTH' else 1)
```
Besides that I manually reviewed all the changes carefully + tested it for a couple of hours. Works fine :ok_hand:

friscoMad commented:
Rebased and removed the lines by @Obihoernchen and the ```cant_reach``` var as I was reviewing that part of the code and was not needed as now we have ```count_late``` to do the same check
